optosim/simulation/event_reader.py

```diff
@@ -71,172 +61,64 @@ def __init__(self, filenames):
-        print("number of files: ", len(self.filenames))
-        self.read_config()
-
-    def read_config(self):
-        """Reads the configuration of the data reader
-
-        Parameters
-        ----------
-        None
-
-        Returns
-        -------
-        None
-
-        A.P. Colijn
-        """
-        if self.nfiles > 0:
-            with h5py.File(self.filenames[0], "r") as hf:
-                self.config = json.loads(hf.attrs["config"])
-
-            hf.close()
-
-    def print_config(self):
-        """Prints the configuration of the data reader
-
-        Parameters
-        ----------
-        None
 
-        Returns
-        -------
-        None
+        # Load config attribute from the first file
+        with h5py.File(filenames[0], 'r') as f:
+            self.config = json.loads(f.attrs.get("config"))
 
-        A.P. Colijn
-        """
-        print(self.config)
-
-    def __iter__(self):
-        """Returns an iterator for the data reader
-
-        Parameters
-        ----------
-        None
-
-        Returns
-        -------
-        self : EventReader
-            The iterator
-
-        A.P. Colijn
-        """
-        return self
-
-    def __next__(self):
-        """Reads the next event from the file
-
-        Parameters
-        ----------
-        None
-
-        Returns
-        -------
-        event_data : dict
-            The event data
-
-        A.P. Colijn
-        """
+        print("huh config = ", self.config)        
 
         if self.config['data_type_version'] == 1.0:
-            while True:
-                # Check if it's time to load or switch to a new file
-                if self.file is None or self.event_index >= len(self.event_names) - 1:
-                    # Close current file if it exists
-                    if self.file:
-                        self.file.close()
-
-                    # Check if we're out of files to read from
-                    if self.file_index >= len(self.filenames):
-                        raise StopIteration
-
-                    # Open next file
-                    self.file = h5py.File(self.filenames[self.file_index], "r")
-                    self.file_index += 1
-                    self.event_names = list(self.file["events"].keys())
-                    self.event_index = -1
-
-                self.event_index += 1
-                event_dataset = self.file["events"][self.event_names[self.event_index]]
-                event_data = {key: event_dataset[key][()] for key in event_dataset.keys()}
-                return event_data
+            # throw an error that this version is obsolete
+            raise ValueError("Data type version 1.0 is obsolete. Please use version 2.0")
+
+        # Load data from all files
+        self.data_dict = self._load_data(filenames)
+        self.number_of_files = len(filenames)
+        self.num_events = len(next(iter(self.data_dict.values())))  # assuming all datasets have the same length
```

Comment:
Is there a better way to do this, without assuming all datasets are the same length?